src/main/java/frc/robot/subsystems/Shooter.java

```diff
@@ -34,106 +36,238 @@
+        System.out.println(shooterRpms);
+        updateManualRPMS();
+
         if (oi.runFlyWheelButtonManual()) {
+            if (checkVision()) {
+                vc.shooterstate = shooterState.ALIGN;
+            }
+
             startShooter(shooterRpms);
-        } else {
+        }
+
+        else if (oi.autoShootButton()) {
+            if (checkVision()) {
+                autoShoot();
+            }
+
+            else {
+                if (checkVision()) {
+                    vc.shooterstate = shooterState.ALIGN;
+                }
+                startShooter(shooterRpms);
+                feederMain();
+            }
+        }
+
+        else {
+            if (checkVision()) {
+                vc.shooterstate = shooterState.ALIGN;
+            }
+
             stopShooter();
         }
+    }
 
-        // Control for lowering intake
-        if (oi.manualIntakeButton() && state == gathererUp) {
-            lowerIntake();
-            startIntake();
-        } else if (!oi.manualIntakeButton() && state == gathererDown) {
-            stopIntake();
-        } else if (oi.manualIntakeButton() && state == holding) {
-            stopIntake();
-            raiseIntake();
+    public void feederMain() {
+        if (!oi.autoShootButton()) {
+            if (oi.shootButton()) {
+                startFeeder(feederSpeed);
+                startIntake(-intakeSpeed);
+            }
+
+            else if (oi.reverseIntake()) {
+                startFeeder(-feederSpeed);
+            }
+
+            else {
+                stopFeeder();
+            }
         }
+    }
+
+    public void startFeeder(double speed) {
+        feeder.set(speed);
+    }
 
+    public void stopFeeder() {
+        feeder.set(0);
     }
 
     public void startShooter(double rpms) {
-        shooter.set(TalonFXControlMode.Velocity, rpms);
-        if (oi.shootButton()) {
-            feeder.set(feederSpeed);
-        }
+        shooter.set(TalonFXControlMode.Velocity, rpms * 2048 / 10); // might need to multiply by 60
+    }
 
-        else {
-            feeder.set(0);
-        }
+    public double getShooterRpms() {
+        return shooter.getSelectedSensorVelocity() / 2048 * 10 / 60;
     }
 
     public void stopShooter() {
         shooter.set(TalonFXControlMode.PercentOutput, 0);
     }
 
     public void lowerIntake() {
-        state = gathererDown;
         lowerIntake.set(true);
+        gathererState = gathererDown;
     }
 
     public void raiseIntake() {
-        state = gathererUp;
         lowerIntake.set(false);
+        gathererState = gathererUp;
     }
 
-    public void startIntake() {
-        intake.set(TalonSRXControlMode.PercentOutput, intakeSpeed);
+    public void startIntake(double speed) {
+        intake.set(TalonSRXControlMode.PercentOutput, speed);
     }
 
     public void stopIntake() {
-        state = holding;
         intake.set(TalonSRXControlMode.PercentOutput, 0);
+        gathererState = holding;
```

Comment:
This probably isnt' the right place to change to holding state, since we stop the intake in different states